warhorn/warhorn.py

```diff
@@ -1198,12 +1181,25 @@ def assemble_warrior():
 
     check_basic_requirements(logfile, config_file_name, console_log_name,
                              print_log_name, python_executable)
-    get_dependencies(logfile, print_log_name, config_file_name)
+    node = get_node(config_file_name, 'virtualenv')
+    if node is not False and get_attribute_value(node, 'activate') == 'yes':
+        if get_attribute_value(node, 'install') == 'yes':
+            install_depen('virtualenv==15.1.0', 'virtualenv', logfile, print_log_name)
+        war_tag = get_node(config_file_name, 'warriorframework')
+        if war_tag is False:
+            print_error("warriorframework is a mandatory repo. Please add and rerun",
+                        logfile, print_log_name)
+            setDone(1)
+            getDone(logfile, print_log_name)
+        dest = get_attribute_value(war_tag, 'destination')
+        virtualenv_activated = activate_virtualenv(node, dest, logfile, print_log_name)
```

Comment:
But, activating virtualenv is with in warhorn script, which would get cleaned up automatically, when you come out of the script.  We can keep the sub-process active if it can be back ground process, but here it can only be a foreground process and when the parent process completes, it would clean up the child processes as well along with the virtualenv environment. Can you elaborate on what you have in mind?